README.md

```diff
@@ -117,47 +120,171 @@ If you'd like to get a message when the button is pressed or released, call the
+section 6.0 APPLICATIONS INFORMATION.  The section goes into greater detail, 
+but figure 6.1 tell us how to modify the request/response to resolve the issue.
+
+![MCP3008 Data Sheet figure 6.1](assets/images/mcp3008-figure-6-1.PNG)
+
+The MCU Transmitted Data row tells us to send 7 `0` bits before the start 
+bit.  Combined with our 17 bit request/response communication, this will give us a 
+total of 24 bits and 3 bytes.  We can see from the MCU Received Data row we'll 
+need to take those 7 bits into account when we receive the data, but we'll also 
+receive 24 bits or 3 bytes.
+
+Using the protocol from the data sheet here is how we will read the moisture 
+sensor.
 
     # Make sure that you've enabled or loaded the SPI driver or this will
     # fail.
     iex> alias ElixirALE.SPI
     iex> {:ok, pid} = SPI.start_link("spidev0.0")
     {:ok, #PID<0.124.0>}
 
-    # Read the potentiometer
+    # Read the moisture sensor
+
+    # Use binary pattern matching to pull out the ADC counts (low 10 bits)
+    iex> mode = 1 #Single-ended mode
+    iex> sensor = 2 #Try the CH2 sensor
+    iex> <<_::size(14), value::size(10)>> =
+           SPI.transfer(pid, <<0x01, mode::size(1), sensor::size(3), 0::size(12)>>)
+    TODO "NEED WHAT IT RETURNS"
+
+    iex> IO.puts "Sensor #{sensor} measured #{value}"
+    TODO "NEED WHAT IT RETURNS"
+
+If you are new to using Elixir for embedded, there may be some things here you 
+might not be familiar with.  We'll be working with bit strings.  If this is 
+new to you, you might want to refer to the 
+[Elixir docs for bitstings](https://hexdocs.pm/elixir/Kernel.SpecialForms.html#%3C%3C%3E%3E/1).
+
+First we call SPI.start_link with Linux spidev device driver as it argument.  
+We pattern match with `{:ok, pid}` so we can use `pid` in the 
+SPI.transfer call next.
+
+    {:ok, pid} = SPI.start_link("spidev0.0")
+
+Elixir ALE uses the Linux spidev device driver for communicating with SPI 
+devices. It creates files in /dev for each SPI interface that it knows about. 
+Many boards only have one SPI interface and it shows up as /dev/spidev0.0. 
+The easiest way to find out the right name is to run ls spidev* in the /dev 
+directory. If nothing shows up, it is likely that the spidev driver needs to 
+be loaded. Try running modprobe spidev or consult your board's documentation.
+
+Next we'll make a call to SPI.transfer to read the moisture sensor.
+
+    SPI.transfer(pid, <<0x01, mode::size(1), sensor::size(3), 0::size(12)>>)
+
+We are building the bitstring in 4 segments.  The first segement `0x01`, is the 
+hexadecimal representation of `00000001` that we determined above we need to 
+send in the first byte.  The final bit `1` will be the start bit.  It is 
+common to describe bytes using hexadecimal notation.
+
+The next bit will be for the mode which we have set to 1 for Single-ended mode.
+
+TODO:  The slides say 0 for Single-ended mode, but that doesnt' seem to match 
```

Comment:
Oops - I reviewed this forgetting that I test Elixir ALE against an MCP3002 and not an MCP3008. That's why things are different. I have a ton of MCP3002s. I'm going to contact you on Slack about getting some parts to you so that we can be documenting and testing against the same thing.